README.md

```diff
@@ -157,10 +157,8 @@ associated to `linux` and `apache-servers` subscriptions.
 
 ```puppet
   class { 'sensu::agent':
-    backends    => ['sensu-backend.example.com:8081'],
-    config_hash => {
```

Comment:
Does this bring us closer to using the same options as Sensu Go or further?

Reply:
Neither? This just makes "important" or more commonly changed agent configs actual parameters instead of requiring everything be in `config_hash`. Several places in docs put emphasis on things like `subscriptions` and also this makes it easier to set values that are sent to the backend entity for the agent, like `labes` and `annotations` as well as `subscriptions`.

Reply:
If we are moving this way in the new version, perhaps we should wait until then as this might confuse people.

Reply:
So save merge till v4?

Reply:
Yes. Want to also ensure that as a design decision we are mimicking the data structures used by sensu-go. We started down the road of making our own and mapping them back in order to make things easy to consume and it led to a lot of drift and our need for this major version bump.

Reply:
Before it was possible to define something like `config_hash` with `subscriptions` that was invalid. I don't think it's realistic to expose all config options but this exposes the important ones and adds extra validation ensuring the correct data types are used.